In [83]:
import pandas as pd

In [84]:
#import csv
df = pd.read_csv('Cricket.csv')

In [85]:
#show the first 20 rows of our dataframe
df.head(20)

Unnamed: 0 Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4  \
0                        NaN        NaN        NaN        NaN        NaN   
1                        NaN        NaN        NaN        NaN        NaN   
2                        NaN        NaN        NaN        NaN        NaN   
3                        NaN        NaN        NaN        NaN        NaN   
4                        NaN        NaN        NaN        NaN        NaN   
5                        NaN        NaN        NaN        NaN        NaN   
6                        NaN        NaN        NaN        NaN        NaN   
7                        NaN        NaN        NaN        NaN        NaN   
8                        NaN        NaN        NaN        NaN        NaN   
9                        NaN        NaN        NaN        NaN        NaN   
10                       NaN        NaN        NaN        NaN        NaN   
11                       NaN        NaN        NaN        NaN        NaN   
12                       NaN        NaN        NaN        NaN        NaN   
13  Highest averages players       Span        Mat       Inns         NO   
14          DG Bradman (AUS)  1928-1948         52         80         10   
15            HC Brook (ENG)  2022-2023         12         20          1   
16            AC Voges (AUS)  2015-2016         20         31          7   
17           RG Pollock (SA)  1963-1970         23         41          4   
18           GA Headley (WI)  1930-1954         22         40          4   
19         H Sutcliffe (ENG)  1924-1935         54         84          9   

   Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  Unnamed: 10  \
0         NaN        NaN        NaN        NaN        NaN          NaN   
1         NaN        NaN        NaN        NaN        NaN          NaN   
2         NaN        NaN        NaN        NaN        NaN          NaN   
3         NaN        NaN        NaN        NaN        NaN          NaN   
4         NaN        NaN        NaN        NaN        NaN          NaN   
5         NaN        NaN        NaN        NaN        NaN          NaN   
6         NaN        NaN        NaN        NaN        NaN          NaN   
7         NaN        NaN        NaN        NaN        NaN          NaN   
8         NaN        NaN        NaN        NaN        NaN          NaN   
9         NaN        NaN        NaN        NaN        NaN          NaN   
10        NaN        NaN        NaN        NaN        NaN          NaN   
11        NaN        NaN        NaN        NaN        NaN          NaN   
12        NaN        NaN        NaN        NaN        NaN          NaN   
13       Runs         HS        Ave         BF         SR        100.0   
14       6996        334      99.94      9800+       58.6         29.0   
15       1181        186      62.15       1287      91.76          4.0   
16       1485       269*      61.87       2667      55.68          5.0   
17       2256        274      60.97      1707+      54.48          7.0   
18       2190       270*      60.83       416+         56         10.0   
19       4555        194      60.73      6558+      34.59         16.0   

    Unnamed: 11  Unnamed: 12 Unnamed: 13 Unnamed: 14  
0           NaN          NaN         NaN         NaN  
1           NaN          NaN         NaN         NaN  
2           NaN          NaN         NaN         NaN  
3           NaN          NaN         NaN         NaN  
4           NaN          NaN         NaN         NaN  
5           NaN          NaN         NaN         NaN  
6           NaN          NaN         NaN         NaN  
7           NaN          NaN         NaN         NaN  
8           NaN          NaN         NaN         NaN  
9           NaN          NaN         NaN         NaN  
10          NaN          NaN         NaN         NaN  
11          NaN          NaN         NaN         NaN  
12          NaN          NaN         NaN         NaN  
13         50.0          0.0          4s          6s  
14         13.0          7.0        626+ 

In [86]:
# Drop the first 13 rows
df = df.iloc[13:]

# Reset the index if you want a continuous index
df.reset_index(drop=True, inplace=True)

In [87]:
df.head()

Unnamed: 0 Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4  \
0  Highest averages players       Span        Mat       Inns         NO   
1          DG Bradman (AUS)  1928-1948         52         80         10   
2            HC Brook (ENG)  2022-2023         12         20          1   
3            AC Voges (AUS)  2015-2016         20         31          7   
4           RG Pollock (SA)  1963-1970         23         41          4   

  Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  Unnamed: 10  \
0       Runs         HS        Ave         BF         SR        100.0   
1       6996        334      99.94      9800+       58.6         29.0   
2       1181        186      62.15       1287      91.76          4.0   
3       1485       269*      61.87       2667      55.68          5.0   
4       2256        274      60.97      1707+      54.48          7.0   

   Unnamed: 11  Unnamed: 12 Unnamed: 13 Unnamed: 14  
0         50.0          0.0          4s          6s  
1         13.0          7.0        626+           6  
2          7.0          1.0         141          23  
3          4.0          2.0         186           5  
4         11.0          1.0        246+          11

In [88]:
# Print column names as a list
print(df.columns.tolist())

['Unnamed: 0', 'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14']


In [89]:
#Rename the column names 
# Create a dictionary to map old column names to new column names
new_column_names = {'Unnamed: 0': 'Player', 
                    'Unnamed: 1': 'Span',
                   'Unnamed: 2': 'Matches',
                   'Unnamed: 3': 'Inns',
                   'Unnamed: 4': 'Not_Outs',
                   'Unnamed: 5': 'Runs',
                   'Unnamed: 6': 'Highest_Inns_Score',
                   'Unnamed: 7': 'Ave',
                   'Unnamed: 8': 'Balls_Faced',
                   'Unnamed: 9': 'Batting_Strike_Rate',
                   'Unnamed: 10': '100.0',
                   'Unnamed: 11': '50.0',
                   'Unnamed: 12': '0.0',
                   'Unnamed: 13': '4s',
                   'Unnamed: 14': '6s',}

# Use the rename method to rename the columns
df.rename(columns=new_column_names, inplace=True)

In [90]:
df

Player       Span Matches  Inns Not_Outs  Runs  \
0   Highest averages players       Span     Mat  Inns       NO  Runs   
1           DG Bradman (AUS)  1928-1948      52    80       10  6996   
2             HC Brook (ENG)  2022-2023      12    20        1  1181   
3             AC Voges (AUS)  2015-2016      20    31        7  1485   
4            RG Pollock (SA)  1963-1970      23    41        4  2256   
..                       ...        ...     ...   ...      ...   ...   
95                       NaN        NaN     NaN   NaN      NaN   NaN   
96                       NaN        NaN     NaN   NaN      NaN   NaN   
97                       NaN        NaN     NaN   NaN      NaN   NaN   
98                       NaN        NaN     NaN   NaN      NaN   NaN   
99                       NaN        NaN     NaN   NaN      NaN   NaN   

   Highest_Inns_Score    Ave Balls_Faced Batting_Strike_Rate  100.0  50.0  \
0                  HS    Ave          BF                  SR  100.0  50.0   
1                 334  99.94       9800+                58.6   29.0  13.0   
2                 186  62.15        1287               91.76    4.0   7.0   
3                269*  61.87        2667               55.68    5.0   4.0   
4                 274  60.97       1707+               54.48    7.0  11.0   
..                ...    ...         ...                 ...    ...   ...   
95                NaN    NaN         NaN                 NaN    NaN   NaN   
96                NaN    NaN         NaN                 NaN    NaN   NaN   
97                NaN    NaN         NaN                 NaN    NaN   NaN   
98                NaN    NaN         NaN                 NaN    NaN   NaN   
99                NaN    NaN         NaN                 NaN    NaN   NaN   

    0.0    4s   6s  
0   0.0    4s   6s  
1   7.0  626+    6  
2   1.0   141   23  
3   2.0   186    5  
4   1.0  246+   11  
..  ...   ...  ...  
95  NaN   NaN  NaN  
96  NaN   NaN  NaN  
97  NaN   NaN  NaN  
98  NaN   NaN  NaN  
99  NaN   NaN  NaN  

[100 rows x 15 columns]

In [91]:
# Drop the first row (index 0)
df = df.drop(0)

# Reset the index if you want a continuous index
df.reset_index(drop=True, inplace=True)

In [92]:
df

Player       Span Matches Inns Not_Outs  Runs  \
0   DG Bradman (AUS)  1928-1948      52   80       10  6996   
1     HC Brook (ENG)  2022-2023      12   20        1  1181   
2     AC Voges (AUS)  2015-2016      20   31        7  1485   
3    RG Pollock (SA)  1963-1970      23   41        4  2256   
4    GA Headley (WI)  1930-1954      22   40        4  2190   
..               ...        ...     ...  ...      ...   ...   
94               NaN        NaN     NaN  NaN      NaN   NaN   
95               NaN        NaN     NaN  NaN      NaN   NaN   
96               NaN        NaN     NaN  NaN      NaN   NaN   
97               NaN        NaN     NaN  NaN      NaN   NaN   
98               NaN        NaN     NaN  NaN      NaN   NaN   

   Highest_Inns_Score    Ave Balls_Faced Batting_Strike_Rate  100.0  50.0  \
0                 334  99.94       9800+                58.6   29.0  13.0   
1                 186  62.15        1287               91.76    4.0   7.0   
2                269*  61.87        2667               55.68    5.0   4.0   
3                 274  60.97       1707+               54.48    7.0  11.0   
4                270*  60.83        416+                  56   10.0   5.0   
..                ...    ...         ...                 ...    ...   ...   
94                NaN    NaN         NaN                 NaN    NaN   NaN   
95                NaN    NaN         NaN                 NaN    NaN   NaN   
96                NaN    NaN         NaN                 NaN    NaN   NaN   
97                NaN    NaN         NaN                 NaN    NaN   NaN   
98                NaN    NaN         NaN                 NaN    NaN   NaN   

    0.0    4s   6s  
0   7.0  626+    6  
1   1.0   141   23  
2   2.0   186    5  
3   1.0  246+   11  
4   2.0  104+    1  
..  ...   ...  ...  
94  NaN   NaN  NaN  
95  NaN   NaN  NaN  
96  NaN   NaN  NaN  
97  NaN   NaN  NaN  
98  NaN   NaN  NaN  

[99 rows x 15 columns]

In [93]:
#Check null Values
df.isnull().any()

Player                 True
Span                   True
Matches                True
Inns                   True
Not_Outs               True
Runs                   True
Highest_Inns_Score     True
Ave                    True
Balls_Faced            True
Batting_Strike_Rate    True
100.0                  True
50.0                   True
0.0                    True
4s                     True
6s                     True
dtype: bool

In [94]:
# Drop rows with NaN values
df = df.dropna()

In [95]:
df

Player       Span Matches Inns Not_Outs  Runs  \
0    DG Bradman (AUS)  1928-1948      52   80       10  6996   
1      HC Brook (ENG)  2022-2023      12   20        1  1181   
2      AC Voges (AUS)  2015-2016      20   31        7  1485   
3     RG Pollock (SA)  1963-1970      23   41        4  2256   
4     GA Headley (WI)  1930-1954      22   40        4  2190   
..                ...        ...     ...  ...      ...   ...   
64   KD Walters (AUS)  1965-1981      74  125       14  5357   
65  GC Smith (ICC/SA)  2002-2014     117  205       13  9265   
66  WH Ponsford (AUS)  1924-1934      29   48        4  2122   
67    SJ McCabe (AUS)  1930-1938      39   62        5  2748   
68   DR Jardine (ENG)  1928-1934      22   33        6  1296   

   Highest_Inns_Score    Ave Balls_Faced Batting_Strike_Rate  100.0  50.0  \
0                 334  99.94       9800+                58.6   29.0  13.0   
1                 186  62.15        1287               91.76    4.0   7.0   
2                269*  61.87        2667               55.68    5.0   4.0   
3                 274  60.97       1707+               54.48    7.0  11.0   
4                270*  60.83        416+                  56   10.0   5.0   
..                ...    ...         ...                 ...    ...   ...   
64                250  48.26       8662+               49.16   15.0  33.0   
65                277  48.25       15525               59.67   27.0  38.0   
66                266  48.22       3118+               44.77    7.0   6.0   
67                232  48.21       3217+               60.02    6.0  13.0   
68                127     48       2110+               25.59    1.0  10.0   

     0.0    4s  6s  
0    7.0  626+   6  
1    1.0   141  23  
2    2.0   186   5  
3    1.0  246+  11  
4    2.0  104+   1  
..   ...   ...  ..  
64   4.0  525+  23  
65  11.0  1165  24  
66   1.0  119+   0  
67   4.0  241+  5+  
68   2.0   53+   0  

[69 rows x 15 columns]

In [96]:
# Check for duplicate rows
duplicates = df.duplicated()

# Show the rows with duplicates
print(df[duplicates])

                Player       Span Matches Inns Not_Outs   Runs  \
12      GS Sobers (WI)  1954-1974      93  160       21   8032   
16      JB Hobbs (ENG)  1908-1930      61  102        7   5410   
38  S Chanderpaul (WI)  1994-2015     164  280       49  11867   
41      SR Waugh (AUS)  1985-2004     168  260       46  10927   
47       JE Root (ENG)  2012-2023     135  247       20  11416   

   Highest_Inns_Score    Ave Balls_Faced Batting_Strike_Rate  100.0  50.0  \
12               365*  57.78       4063+               53.58   26.0  30.0   
16                211  56.94       5363+               46.22   15.0  28.0   
38               203*  51.37       27395               43.31   30.0  66.0   
41                200  51.06       22461               48.64   32.0  50.0   
47                254  50.29       20113               56.75   30.0  60.0   

     0.0    4s   6s  
12  12.0  593+  32+  
16   4.0  276+    8  
38  15.0  1285   36  
41  22.0  1175   20  
47  12.0  1247   43  


In [97]:
#Drop duplicates
df = df.drop_duplicates()

In [98]:
# Check again for duplicate rows
duplicates = df.duplicated()

# Show the rows with duplicates
print(df[duplicates])

Empty DataFrame
Columns: [Player, Span, Matches, Inns, Not_Outs, Runs, Highest_Inns_Score, Ave, Balls_Faced, Batting_Strike_Rate, 100.0, 50.0, 0.0, 4s, 6s]
Index: []


In [99]:
#Check null Values
df.isnull().any()

Player                 False
Span                   False
Matches                False
Inns                   False
Not_Outs               False
Runs                   False
Highest_Inns_Score     False
Ave                    False
Balls_Faced            False
Batting_Strike_Rate    False
100.0                  False
50.0                   False
0.0                    False
4s                     False
6s                     False
dtype: bool

### Split the span into start and end date


In [100]:
df['Span'].str.split(pat= '-')

0     [1928, 1948]
1     [2022, 2023]
2     [2015, 2016]
3     [1963, 1970]
4     [1930, 1954]
          ...     
64    [1965, 1981]
65    [2002, 2014]
66    [1924, 1934]
67    [1930, 1938]
68    [1928, 1934]
Name: Span, Length: 64, dtype: object

In [101]:
df['Debut_Year']= df['Span'].str.split(pat = '-').str[0]

/var/folders/ds/324bd2qs425cy9z13chd2xx80000gn/T/ipykernel_1468/689157003.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Debut_Year']= df['Span'].str.split(pat = '-').str[0]


In [102]:
df['Final_Year']= df['Span'].str.split(pat = '-').str[1]

/var/folders/ds/324bd2qs425cy9z13chd2xx80000gn/T/ipykernel_1468/962401666.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Final_Year']= df['Span'].str.split(pat = '-').str[1]


In [103]:
df

Player       Span Matches Inns Not_Outs  Runs  \
0    DG Bradman (AUS)  1928-1948      52   80       10  6996   
1      HC Brook (ENG)  2022-2023      12   20        1  1181   
2      AC Voges (AUS)  2015-2016      20   31        7  1485   
3     RG Pollock (SA)  1963-1970      23   41        4  2256   
4     GA Headley (WI)  1930-1954      22   40        4  2190   
..                ...        ...     ...  ...      ...   ...   
64   KD Walters (AUS)  1965-1981      74  125       14  5357   
65  GC Smith (ICC/SA)  2002-2014     117  205       13  9265   
66  WH Ponsford (AUS)  1924-1934      29   48        4  2122   
67    SJ McCabe (AUS)  1930-1938      39   62        5  2748   
68   DR Jardine (ENG)  1928-1934      22   33        6  1296   

   Highest_Inns_Score    Ave Balls_Faced Batting_Strike_Rate  100.0  50.0  \
0                 334  99.94       9800+                58.6   29.0  13.0   
1                 186  62.15        1287               91.76    4.0   7.0   
2                269*  61.87        2667               55.68    5.0   4.0   
3                 274  60.97       1707+               54.48    7.0  11.0   
4                270*  60.83        416+                  56   10.0   5.0   
..                ...    ...         ...                 ...    ...   ...   
64                250  48.26       8662+               49.16   15.0  33.0   
65                277  48.25       15525               59.67   27.0  38.0   
66                266  48.22       3118+               44.77    7.0   6.0   
67                232  48.21       3217+               60.02    6.0  13.0   
68                127     48       2110+               25.59    1.0  10.0   

     0.0    4s  6s Debut_Year Final_Year  
0    7.0  626+   6       1928       1948  
1    1.0   141  23       2022       2023  
2    2.0   186   5       2015       2016  
3    1.0  246+  11       1963       1970  
4    2.0  104+   1       1930       1954  
..   ...   ...  ..        ...        ...  
64   4.0  525+  23       1965       1981  
65  11.0  1165  24       2002       2014  
66   1.0  119+   0       1924       1934  
67   4.0  241+  5+       1930       1938  
68   2.0   53+   0       1928       1934  

[64 rows x 17 columns]

### Drop Span Columns

In [104]:
df = df.drop(['Span'], axis =1)

### Split the country from the player

In [105]:
df['Player'].str.split(pat = '(')

0      [DG Bradman , AUS)]
1        [HC Brook , ENG)]
2        [AC Voges , AUS)]
3       [RG Pollock , SA)]
4       [GA Headley , WI)]
              ...         
64     [KD Walters , AUS)]
65    [GC Smith , ICC/SA)]
66    [WH Ponsford , AUS)]
67      [SJ McCabe , AUS)]
68     [DR Jardine , ENG)]
Name: Player, Length: 64, dtype: object

In [106]:
df['Country'] = df['Player'].str.split(pat = '(').str[1]

In [107]:
df['Country'] = df['Country'].str.split(pat = ')').str[0]

In [108]:
df['Country'] 

0        AUS
1        ENG
2        AUS
3         SA
4         WI
       ...  
64       AUS
65    ICC/SA
66       AUS
67       AUS
68       ENG
Name: Country, Length: 64, dtype: object

In [109]:
df['Player'] = df['Player'].str.split(pat = '(').str[0]

In [110]:
df.head()

Player Matches Inns Not_Outs  Runs Highest_Inns_Score    Ave  \
0  DG Bradman       52   80       10  6996                334  99.94   
1    HC Brook       12   20        1  1181                186  62.15   
2    AC Voges       20   31        7  1485               269*  61.87   
3  RG Pollock       23   41        4  2256                274  60.97   
4  GA Headley       22   40        4  2190               270*  60.83   

  Balls_Faced Batting_Strike_Rate  100.0  50.0  0.0    4s  6s Debut_Year  \
0       9800+                58.6   29.0  13.0  7.0  626+   6       1928   
1        1287               91.76    4.0   7.0  1.0   141  23       2022   
2        2667               55.68    5.0   4.0  2.0   186   5       2015   
3       1707+               54.48    7.0  11.0  1.0  246+  11       1963   
4        416+                  56   10.0   5.0  2.0  104+   1       1930   

  Final_Year Country  
0       1948     AUS  
1       2023     ENG  
2       2016     AUS  
3       1970      SA  
4       1954      WI

### Change Datatype


In [111]:
df.dtypes

Player                  object
Matches                 object
Inns                    object
Not_Outs                object
Runs                    object
Highest_Inns_Score      object
Ave                     object
Balls_Faced             object
Batting_Strike_Rate     object
100.0                  float64
50.0                   float64
0.0                    float64
4s                      object
6s                      object
Debut_Year              object
Final_Year              object
Country                 object
dtype: object

### Remove the stars(*) from the stats

In [112]:
df['Highest_Inns_Score'].str.split(pat = '*').str[0]

0     334
1     186
2     269
3     274
4     270
     ... 
64    250
65    277
66    266
67    232
68    127
Name: Highest_Inns_Score, Length: 64, dtype: object

In [113]:
df['Highest_Inns_Score'] = df['Highest_Inns_Score'].str.split(pat = '*').str[0]

In [114]:
df['Highest_Inns_Score']

0     334
1     186
2     269
3     274
4     270
     ... 
64    250
65    277
66    266
67    232
68    127
Name: Highest_Inns_Score, Length: 64, dtype: object

In [115]:
df['Highest_Inns_Score'] = df['Highest_Inns_Score'].astype('int')

In [116]:
df.dtypes

Player                  object
Matches                 object
Inns                    object
Not_Outs                object
Runs                    object
Highest_Inns_Score       int64
Ave                     object
Balls_Faced             object
Batting_Strike_Rate     object
100.0                  float64
50.0                   float64
0.0                    float64
4s                      object
6s                      object
Debut_Year              object
Final_Year              object
Country                 object
dtype: object

### Fix the datatypes of Debut year and the final year 

In [117]:
df = df.astype({'Debut_Year':'int',
               'Final_Year':'int',
               })

In [118]:
df['Matches'] = df['Matches'].str.split(pat = '*').str[0]

In [119]:
df['Matches'] = df['Matches'].astype('int')

In [120]:
df.dtypes

Player                  object
Matches                  int64
Inns                    object
Not_Outs                object
Runs                    object
Highest_Inns_Score       int64
Ave                     object
Balls_Faced             object
Batting_Strike_Rate     object
100.0                  float64
50.0                   float64
0.0                    float64
4s                      object
6s                      object
Debut_Year               int64
Final_Year               int64
Country                 object
dtype: object

### Fix the datatypes of Balls_faced, Batting_strike_faced & the rest 


In [121]:
df['Balls_Faced'] = df['Balls_Faced'].str.split(pat = '+').str[0]

In [122]:
df['Balls_Faced'] = df['Balls_Faced'].astype('int')

ValueError: invalid literal for int() with base 10: '-'

In [123]:
# Replace '-' with 0
df = df.replace('-', 0)

In [124]:
df['Balls_Faced'] = df['Balls_Faced'].astype('int')

In [125]:
df['Batting_Strike_Rate'] = df['Batting_Strike_Rate'].astype('float')

In [126]:
df['Inns'] = df['Inns'].astype('int')

In [127]:
df['Not_Outs'] = df['Not_Outs'].astype('int')

In [128]:
df['Runs'] = df['Runs'].astype('int')

In [129]:
df['Ave'] = df['Ave'].astype('float')

In [130]:
df['100.0'] = df['100.0'].astype('int')

In [131]:
df['50.0'] = df['50.0'].astype('int')

In [132]:
df['0.0'] = df['0.0'].astype('int')

In [133]:
df['4s'] = df['4s'].str.split(pat = '+').str[0]

In [134]:
df['4s'] = df['4s'].astype('int')

In [135]:
df['6s'] = df['6s'].str.split(pat = '+').str[0]

In [136]:
df['6s'] = df['6s'].astype('int')

In [137]:
df.dtypes

Player                  object
Matches                  int64
Inns                     int64
Not_Outs                 int64
Runs                     int64
Highest_Inns_Score       int64
Ave                    float64
Balls_Faced              int64
Batting_Strike_Rate    float64
100.0                    int64
50.0                     int64
0.0                      int64
4s                       int64
6s                       int64
Debut_Year               int64
Final_Year               int64
Country                 object
dtype: object

### Create a career length column

In [138]:
df['Career_length'] = (df['Final_Year'] - df['Debut_Year'])

In [139]:
df

Player  Matches  Inns  Not_Outs  Runs  Highest_Inns_Score    Ave  \
0    DG Bradman        52    80        10  6996                 334  99.94   
1      HC Brook        12    20         1  1181                 186  62.15   
2      AC Voges        20    31         7  1485                 269  61.87   
3    RG Pollock        23    41         4  2256                 274  60.97   
4    GA Headley        22    40         4  2190                 270  60.83   
..           ...      ...   ...       ...   ...                 ...    ...   
64   KD Walters        74   125        14  5357                 250  48.26   
65     GC Smith       117   205        13  9265                 277  48.25   
66  WH Ponsford        29    48         4  2122                 266  48.22   
67    SJ McCabe        39    62         5  2748                 232  48.21   
68   DR Jardine        22    33         6  1296                 127  48.00   

    Balls_Faced  Batting_Strike_Rate  100.0  50.0  0.0    4s  6s  Debut_Year  \
0          9800                58.60     29    13    7   626   6        1928   
1          1287                91.76      4     7    1   141  23        2022   
2          2667                55.68      5     4    2   186   5        2015   
3          1707                54.48      7    11    1   246  11        1963   
4           416                56.00     10     5    2   104   1        1930   
..          ...                  ...    ...   ...  ...   ...  ..         ...   
64         8662                49.16     15    33    4   525  23        1965   
65        15525                59.67     27    38   11  1165  24        2002   
66         3118                44.77      7     6    1   119   0        1924   
67         3217                60.02      6    13    4   241   5        1930   
68         2110                25.59      1    10    2    53   0        1928   

    Final_Year Country  Career_length  
0         1948     AUS             20  
1         2023     ENG              1  
2         2016     AUS              1  
3         1970      SA              7  
4         1954      WI             24  
..         ...     ...            ...  
64        1981     AUS             16  
65        2014  ICC/SA             12  
66        1934     AUS             10  
67        1938     AUS              8  
68        1934     ENG              6  

[64 rows x 18 columns]

# Questions

### 1. What is the average career length?

In [141]:
df['Career_length'].mean()

12.75

The average career length is 12.75 years

### 2. What is the average Batting_strike_rate for cricketers who played over 10 years?

In [143]:
df[df['Career_length'] > 10]['Batting_Strike_Rate'].mean()

47.95454545454545

### 3. Find number of cricketers who played before 1960 

In [149]:
df[df['Debut_Year'] <1960]['Player'].count()

23

### 4. Max highest Inns score by country

In [156]:
df.groupby('Country')['Highest_Inns_Score'].max().to_frame('Highest In Country').reset_index().sort_values('Highest In Country', ascending = False)

Country  Highest In Country
5    ICC/WI                 400
0       AUS                 380
10       SL                 374
11       WI                 365
1       ENG                 364
3   ICC/PAK                 329
2   ICC/IND                 319
8       PAK                 313
9        SA                 278
4    ICC/SA                 277
6       IND                 254
7        NZ                 251
12      ZIM                 232

### 5. Hundreds, Fifties, ducks (0) AVG by country

In [159]:
df.groupby('Country')[['100.0','50.0','0.0']].mean()

100.0       50.0        0.0
Country                                 
AUS      20.625000  28.000000   8.500000
ENG      12.307692  20.769231   4.307692
ICC/IND  29.500000  47.500000  12.000000
ICC/PAK  25.000000  46.000000  15.000000
ICC/SA   36.000000  48.000000  13.500000
ICC/WI   34.000000  48.000000  17.000000
IND      29.500000  36.250000  10.750000
NZ       12.333333  16.333333   4.333333
PAK      17.800000  23.200000   8.000000
SA        9.800000  20.200000   3.400000
SL       28.666667  44.000000  12.333333
WI       16.625000  25.625000   7.250000
ZIM      12.000000  27.000000   5.000000

In [160]:
# Export the DataFrame to a new CSV file
df.to_csv('Final_Cleaned_Data.csv', index=False)